invokeai/app/services/workflow_records/workflow_records_common.py

```diff
@@ -1,2 +1,103 @@
+import datetime
+from enum import Enum
+from typing import Any, Union
+
+import semver
+from pydantic import BaseModel, Field, JsonValue, TypeAdapter, field_validator
+
+from invokeai.app.util.metaenum import MetaEnum
+from invokeai.app.util.misc import uuid_string
+
+__workflow_meta_version__ = semver.Version.parse("1.0.0")
+
+
+class ExposedField(BaseModel):
+    nodeId: str
+    fieldName: str
+
+
 class WorkflowNotFoundError(Exception):
     """Raised when a workflow is not found"""
+
+
+class WorkflowRecordOrderBy(str, Enum, metaclass=MetaEnum):
+    """The order by options for workflow records"""
+
+    CreatedAt = "created_at"
+    UpdatedAt = "updated_at"
+    OpenedAt = "opened_at"
+    Name = "name"
+
+
+class WorkflowCategory(str, Enum, metaclass=MetaEnum):
```

Comment:
Will commercial be able to override `WorkflowCategory` and `WorkflowRecordOrderBy`? And does the UI use these to decide what options to render?

Reply:
We decided to wait to make changes here, there's no db schema enforcing the valid values so we can just update the app if we want to extra categories et